app/uk/gov/hmrc/thirdpartyapplication/services/commands/ChangeResponsibleIndividualCommandHandler.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.thirdpartyapplication.services.commands
+
+import cats.Apply
+import cats.data.{NonEmptyChain, NonEmptyList, Validated, ValidatedNec}
+import uk.gov.hmrc.apiplatform.modules.submissions.domain.models.Submission
+import uk.gov.hmrc.apiplatform.modules.submissions.services.SubmissionsService
+import uk.gov.hmrc.thirdpartyapplication.domain.models.{ChangeResponsibleIndividual, ImportantSubmissionData, Standard, UpdateApplicationEvent}
+import uk.gov.hmrc.thirdpartyapplication.models.db.ApplicationData
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class ChangeResponsibleIndividualCommandHandler @Inject()(
+    submissionService: SubmissionsService
+  )(implicit val ec: ExecutionContext
+  ) extends CommandHandler {
+
+  import CommandHandler._
+
+  private def isNotCurrentRi(name: String, email: String, app: ApplicationData) =
+    cond(app.access match {
+      case Standard(_, _, _, _, _, Some(ImportantSubmissionData(_, responsibleIndividual, _, _, _, _))) =>
+        ! responsibleIndividual.fullName.value.equalsIgnoreCase(name) || ! responsibleIndividual.emailAddress.value.equalsIgnoreCase(email)
+      case _ => true
+    }, s"The specified individual $name [$email] is already the RI for this application")
```

Comment:
Is there any chance this could be logged out?
That email looks like PII.

Reply:
I don't think it gets logged, could be wrong though

Reply:
Yeah I just checked the code and can't see how it would be logged.
Best to sanitise it just in case.